lib/utils.js

```diff
@@ -130,6 +130,13 @@ exports.formatNamedParameters = formatNamedParameters;
 function cloneDeep(obj) {
   obj = obj || {};
   return _.cloneDeepWith(obj, elem => {
+    // Special handling for the inclusive array property (for range types)
+    if (Array.isArray(elem) && elem.hasOwnProperty('inclusive')) {
```

Comment:
This will execute for each clone operation I would like to avoid this if possible, there is also one more issue https://github.com/sequelize/sequelize/issues/8471

As you are using this type in real world, what do you think about having range type store values like

```js
[{  
  value: 10,
  inclusive: true
}, {
  value: 10,
  inclusive: true
}]
```

Rather than having non-standard array property?

Reply:
I agree that not having to worry about special handling for the non-standard "inclusive" array property would be much cleaner. We should move to a standard representation of range types for both query predicates and query results ensuring the inclusive property is still maintained when results are converted to JSON.

This will be a breaking change though, as we (and I expect many others) have a lot of code that expects range values returned in query results to be of the form

```
[ 10, 10 ]
```

Another approach might be to allow range types to have a 3rd "options" element in the array. When querying/inserting you could define your range as any of
```
[ 10, 10 ] // Defaults to inclusive [true, false]
[ 10, 10, { inclusive: true }]
[ 10, 10, { inclusive: [ true, false ] }]
```
Query results would always be returned as
```
[ 10, 10, { inclusive: [ true, false ] }]
```
This will still be a breaking change, but may lessen the impact on those queries that currently use ranges in their default form (without being concerned about inclusivity).

Reply:
As we are under v5 beta process I would prefer just do one time switch to proper format, which will solve both [clone](https://github.com/sequelize/sequelize/issues/8176) and [stringify](https://github.com/sequelize/sequelize/issues/8471) issues

```js
[{  
  value: 10,
  inclusive: true
}, {
  value: 10,
  inclusive: true
}]
```

Reply:
No problem. Is it ok if I make these changes under this pull request? There'll be some documentation changes too which I'll try to cover.

Reply:
Yeah no problem